activity/journal.go

```diff
@@ -0,0 +1,67 @@
+package activity
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+)
+
+// Journal contains the command-line used to invoke the process associated with this item
+// and its journal ID.
+type Journal struct {
+	ID, Description string
+	Start           time.Time
```

Comment:
Based on how `Start` is used in the code, I'm guessing that it represents when the activity was first started right? If so, then I don't think it belongs in the `Journal` type because the `Journal` type's meant to be re-created multiple times, while `Start` is not supposed to change (this is especially true in `JournalForPID`, where API clients that use it will find that history will record a different Start time than the one specified in `JournalForPID`.).

I think it's worth creating a separate `historyEntry` type to encapsulate the `Start` time info, and have `Start` time be set when we're adding it to the history.

We could also add it to the history when the journal's first created (and maybe create the journal file along the way so that `history <ix>` won't return an error?).

Reply:
There's multiple roles being served here. Maybe I do need to break them out into separate things.

There's a type that's created from a process ID or server request, that's used to interact with the journal.

I think I can separate those into distinct things.

Reply:
Updated everything else. Still working on this.